[Kernel] Updating to 6.1.31

DT3264 opened the pull request:
[Kernel] Updating to 6.1.31

This PR will include all the required migrations for the kernel in order to support debian 12

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/equetzal/huronOS-build-tools/pull/114).
* #235
* __->__ #114

equetzal commented:
Hey, for this PR maybe it would worth using sapling stack features so that you can split this PR, I have some ideas like:
- 1 PR for the kernel stuff
- 1 PR for the base layer
- For the firmware layer
- For Budgie layer
- Services 
- etc.

The division can be other but it would be nice to split it and the ship stacked!

DT3264 commented:
Relevant issue: #113

DT3264 committed:
[Kernel] Updating to 6.1.31

This PR will include all the required migrations for the kernel in order to support debian 12
